Documentation/storage-providers.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Storage Providers
+weight: 12050
+indent: true
+---
+
+# Storage Providers
+
+Rook is the home for operators for multiple storage providers. Each of these storage providers
+has specific requirements and each of them is very independent. There is no runtime dependency
+between the storage providers. Development is where the storage providers benefit one another.
```

Comment:
Not quite sure whether it should be `benefit one another` or `benefit from one another`?  (Probably just a non-native speaker's lack of feel for the language, I know both make sense, but the latter seemed a bit more natural to me.)

Reply:
It is a very subtle difference, but I agree the latter is better.